### ISSUE TITLE
Implement a real backup/restore mechanism

### ISSUE THREAD

**Issue by barbeau** [OneBusAway/onebusaway-android#489]
**Summary:** 

Our current implementation of backup/restore is simply to copy the database file to storage in https://github.com/OneBusAway/onebusaway-android/blob/master/onebusaway-android/src/main/java/org/onebusaway/android/io/backup/Backup.java#L54.  When we "restore" old data, we swap the old database back in.

This causes issues when trying to restore a database from an older database version with a newer version of the app - see https://github.com/OneBusAway/onebusaway-android/issues/473 and https://github.com/OneBusAway/onebusaway-android/issues/478.

The proper fix for the above issues is toimplement a real backup/restore mechanism that exports/imports the data without copying the entire database file.

We've implemented analytics in https://github.com/OneBusAway/onebusaway-android/pull/475 to see how many people are using this feature to help gauge how important it is to implement a new backup method.

**Steps to reproduce:** 

See https://github.com/OneBusAway/onebusaway-android/issues/473 and https://github.com/OneBusAway/onebusaway-android/issues/478.

**Expected behavior:** 

Export/import the data,without changing the database number and triggering `SQLiteOpenHelper.onDowngrade()`.

**Observed behavior:** 

The database version changes and `SQLiteOpenHelper.onDowngrade()` is triggered.

**Device and Android version:** 

Stock LG G4 w/ Android 6.0

**Comment by barbeau** [comment]
This will probably be related to, or at least overlap in implementation partially, the cloud saves feature (https://github.com/OneBusAway/onebusaway-android/issues/32).

**Comment by bbodenmiller** [comment]
What's the status on this feature? Currently it appears that switching to a new Android device you lose all app settings. It'd be nice if OneBusAway used Google's cloud backup - https://developer.android.com/guide/topics/data/backup.html

**Comment by barbeau** [comment]
Correct, preferences aren't part of our current backup mechanism as they exist outside the database.  I haven't had a chance to look into this - our current backup process works well enough that other issues have been higher priority.

**Comment by bmander** [comment]
Implemented — backup/restore now merges the backup's rows into the live Room DB in a single transaction via `LegacyDataImporter` (`backup/Backup.kt`) rather than swapping the DB file, which is the real export/import mechanism this asked for (and it sidesteps the old `onDowngrade` problem). Closing.
